pkg/dockerregistry/server/blobdescriptorservice.go

```diff
@@ -208,17 +210,10 @@ func imageHasBlob(
 		return true
 	}
 
-	if len(image.DockerImageLayers) == 0 {
-		if len(image.DockerImageManifestMediaType) > 0 {
-			// If the media type is set, we can safely assume that the best effort to fill the image layers
-			// has already been done. There are none.
-			return false
-		}
-		err = imageapi.ImageWithMetadata(image)
```

Comment:
Dropping this `ImageWithMetadata` isn't obvious to me.

@miminar ptal.

Reply:
@deads2k this is more complicated... external Image have everything encoded in JSON inside DockerImageMetadata so the filling happens differently there...